feat:  `vim.inspect_pos()`, `vim.show_pos()` and `:Inspect` (#21393)
neovim/neovim: runtime/doc/lua.txt

        |vim.in_fast_event()|

==============================================================================
Lua module: inspector                                          *lua-inspector*

inspect_pos({bufnr}, {row}, {col}, {filter})               *vim.inspect_pos()*
    Get all the items at a given buffer position Can also be pretty printed
    with *:Inspect!*

    Parameters: ~
      • {bufnr}   buffer|nil defaults to the current buffer

>       • {row}     (number|nil) 0-indexed row. Defaults to the current cursor

Reviewer: To be more consistent with the wording used in api.txt:

```suggestion
      • {row}     (number|nil) row to inspect, 0-based. Defaults to row of the current cursor
```

And likewise below.

      • {col}     (number|nil) 0-indexed col. Defaults to the current cursor
      • {filter}  (table|nil) a table with key-value pairs to filter the items
                  • syntax (boolean): include syntax based highlight groups
                    (defaults to true)
                  • treesitter (boolean): include treesitter based highlight
                    groups (defaults to true)

>                   • extmarks (boolean): include extmarks (defaults to true)
>                   • extmarks_with_hl_group (boolean): when false, also include

Reviewer: I would combine these keys and make the option an enum. Something like:

```suggestion
                  • extmarks (string): include extmarks. One of `all`, `none`, `with_hl`.
```

Author: good idea. Changed to be `true|false|"all"`, to keep it more in line with the other options.

        • true and full path of target file if operation was successful
        • false and error message on failure

==============================================================================
Lua module: inspector                                          *lua-inspector*

get({bufnr}, {row}, {col}, {filter})                     *vim.inspector.get()*

>     Get all the items at a given buffer position

Reviewer 2: Since this is a "get all the things" interface, it can live directly on `vim.inspect_pos()`. Don't need a new "inspector" module.

Another reason that a new module doesn't fit here is, we have various modules lie vim.treesitter, vim.diagnostic, that cover topics, and they would have `vim.x.inspect()` functions on them. So it doesn't make sense to invert that relationship and put things in a `vim.inspector` module.

    Parameters: ~
      • {bufnr}   buffer|nil defaults to the current buffer
      • {row}     (number|nil) 0-indexed row. Defaults to the current cursor
      • {col}     (number|nil) 0-indexed col. Defaults to the current cursor
      • {filter}  (table|nil) a table with key-value pairs to filter the items
                  • syntax (boolean): include syntax based highlight groups
                    (defaults to true)
                  • treesitter (boolean): include treesitter based highlight
                    groups (defaults to true)
                  • extmarks (boolean): include extmarks (defaults to true)
                  • extmarks_with_hl_group (boolean): when false, also include
                    extmarks without a hl_group (defaults to true)
                  • semantic_tokens (boolean): include semantic tokens
                    (defaults to true)

> show({bufnr}, {row}, {col}, {filter})                   *vim.inspector.show()*

Reviewer 2: I think we can eliminate the "show" verb across the entire API; instead the "inspect" was long established already. (Can rename `vim.treesitter.show_tree()` => `vim.treesitter.inspect()`).

- Instead, we can name this `vim.inspect_pos()`
- We probably need to leave the door open for a future `win` parameter. So this should be (only) a kwargs interface.

>     Show all the items at a given buffer position

Reviewer 2: Since this is a "show all the things" interface, it can live directly on vim.show_pos(). Don't need a new "inspector" module.
